### ISSUE TITLE
Tensorflow 2.0 custom_scalar plugin for tensorboard

### ISSUE THREAD

**Issue by PgLoLo** [tensorflow/tensorflow#32918]
To set layout for custom_layout plugin in tensorflow 1.x you should use method add_summary in FileWriter class. So the code should look like this:

```python
with tf.summary.FileWriter(LOGDIR) as writer:
   layout_summary = summary_lib.custom_scalar_pb(...)
   writer.add_summary(layout_summary)
```

In tensorflow 2 add_summary method was removed. How can I set custom_scalar layout now? Is it a bug or a desired API change?

**Comment by stephenjfox** [comment]
You should be able to follow the instruction in [the new `tf.summary` documentation](https://www.tensorflow.org/tensorboard/get_started#using_tensorboard_with_other_methods).

Notedly, the following lines seem to allow you to record whatever metrics you want

```python
# define a metric to track
train_loss = tf.keras.metrics.Mean('train_loss', dtype=tf.float32)
test_loss = tf.keras.metrics.Mean('test_loss', dtype=tf.float32)

# track those metrics
for (x_train, y_train) in train_dataset:
    train_step(model, optimizer, x_train, y_train)
  with train_summary_writer.as_default():
    tf.summary.scalar('loss', train_loss.result(), step=epoch) # this line
    tf.summary.scalar('test_loss', test_loss.result(), step=epoch) # and this line
```

If the `'loss'` and `'test_loss'` strings were `'cats'` and `'dogs'`, the API call would still work (it just wouldn't be as helpful to some).
Consider the [documentation for writing your own `Metric` objects](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/Metric?version=stable) if that's your real need.

Hope this helps.

**Comment by dcbr** [comment]
I managed to make the custom_scalar plugin work in TF2.3 as follows:
```py
import tensorflow as tf
from tensorboard.plugins.custom_scalar import layout_pb2
from tensorboard.plugins.custom_scalar.summary import pb as custom_scalar_pb

writer = tf.summary.create_file_writer(LOGDIR)
layout = custom_scalar_pb(layout_pb2.Layout(...))
with writer.as_default():
    tensor = tf.constant(tf.make_ndarray(layout.value[0].tensor))
    tf.summary.write(layout.value[0].tag, tensor, step=0, metadata=layout.value[0].metadata)
```

It would however be nice to have native TF2 support for this, allowing us to write something along the lines `tf.summary.custom_scalar(layout)`.

**Comment by tilakrayal** [comment]
@PgLoLo ,
Is this still an issue?
Could you please update TensorFlow to the latest stable version v.2.6 and let us know if you are facing the same error. Thanks!
